packages/editor/src/systems/ClickPlacementSystem.tsx

```diff
@@ -0,0 +1,280 @@
+import { RendererComponent } from '@etherealengine/spatial/src/renderer/WebGLRendererSystem'
+import { GroupComponent } from '@etherealengine/spatial/src/renderer/components/GroupComponent'
+import { ObjectLayerComponents } from '@etherealengine/spatial/src/renderer/components/ObjectLayerComponent'
+import { ObjectLayers } from '@etherealengine/spatial/src/renderer/constants/ObjectLayers'
+import { EntityTreeComponent, iterateEntityNode } from '@etherealengine/spatial/src/transform/components/EntityTree'
+import { useEffect } from 'react'
+import { Euler, Material, Mesh, Quaternion, Raycaster, Vector3 } from 'three'
+import { EditorHelperState, PlacementMode } from '../services/EditorHelperState'
+import { ObjectGridSnapState } from './ObjectGridSnapSystem'
+
+import { getModelSceneID } from '@etherealengine/engine/src/scene/functions/loaders/ModelFunctions'
+import { HolographicMaterial } from '@etherealengine/engine/src/scene/materials/constants/material-prototypes/HolographicMaterial.mat'
+import { MeshComponent } from '@etherealengine/spatial/src/renderer/components/MeshComponent'
+import React from 'react'
+
+export const ClickPlacementState = defineState({
+  name: 'ClickPlacementState',
+  initial: {
+    placementEntity: UndefinedEntity as Entity,
+    placedEntity: UndefinedEntity as Entity,
+    selectedAsset: null as string | null,
+    yawOffset: 0,
+    pitchOffset: 0,
+    rollOffset: 0,
+    materialCache: [] as [Mesh, Material][]
+  }
+})
+
+const PlacementModelReactor = (props: { placementEntity: Entity }) => {
+  const clickState = useState(getMutableState(ClickPlacementState))
+  const sceneState = useState(getMutableState(SceneState))
+  const placementModel = useOptionalComponent(props.placementEntity, ModelComponent)
+
+  useEffect(() => {
+    if (!placementModel) return
+    const sceneID = getModelSceneID(props.placementEntity)
+    if (!sceneState.scenes[sceneID]) return
+    iterateEntityNode(props.placementEntity, (entity) => {
+      const mesh = getOptionalComponent(entity, MeshComponent)
+      if (!mesh) return
+      const material = mesh.material as Material
+      clickState.materialCache.set((prev) => [...prev, [mesh, material]])
+      mesh.material = new HolographicMaterial({})
+    })
+  }, [placementModel?.scene, sceneState.scenes.keys])
+
+  return null
+}
+
+const objectLayerQuery = defineQuery([ObjectLayerComponents[ObjectLayers.Scene]])
+
+export const ClickPlacementSystem = defineSystem({
+  uuid: 'ee.studio.ClickPlacementSystem',
+  insert: { before: TransformSystem },
+  reactor: () => {
+    const clickState = useState(getMutableState(ClickPlacementState))
+    const editorState = useState(getMutableState(EditorHelperState))
+    const sceneState = useState(getMutableState(SceneState))
+
+    const renderers = defineQuery([RendererComponent])
+
+    const getParentEntity = () => {
+      if (!sceneState.scenes.value) return null
+      const sceneID = sceneState.scenes.keys[0] as SceneID
```

Comment:
why are we implicitly getting index 0 here? we must be getting the scene explicitly (and for multi-scene support)